### PR TITLE
[quick_actions]make QuickActions testable

### PR DESCRIPTION
**Pull request by The-Redhat**
## Description
I used the same approach like in FirebaseMessaging and added a `QuickActions.private` constructor used for testing with a mock MethodChannel. 

## Related Issues

Fixes [#25887](https://github.com/flutter/flutter/issues/25887)

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by The-Redhat** [comment]
I signed it!

**Comment by The-Redhat** [comment]
Thanks for your answer, what should the unit test consist of ?

**Comment by The-Redhat** [comment]
I  made the requested changes!

**Comment by The-Redhat** [comment]
Hey, can anybody please look into this pr? Thanks!

**Comment by justinmc** [comment]
@The-Redhat Can you merge master into this and fix the conflicts?  Then we can try to get a code review for this and get things moving.

**Comment by The-Redhat** [comment]
merged master

File: packages/quick_actions/lib/quick_actions.dart [hunk 1]
```diff
@@ -7,9 +7,6 @@ import 'dart:async';
 import 'package:flutter/services.dart';
 import 'package:meta/meta.dart';
 
-const MethodChannel _kChannel =
```
Comment (cyanglaz) [inline packages/quick_actions/lib/quick_actions.dart]: Can we still keep this and use this const when we init the QuickAction using default?
Something like 
```dart
static final QuickActions _instance = QuickActions.private(
    const MethodChannel(_kChannel),
  );
```

File: packages/quick_actions/lib/quick_actions.dart [hunk 2]
```diff
@@ -36,7 +33,16 @@ class ShortcutItem {
...
+  factory QuickActions() => _instance;
+
+  @visibleForTesting
+  QuickActions.private(MethodChannel channel) : _kChannel = channel;
+
+  static final QuickActions _instance = QuickActions.private(
+    const MethodChannel('plugins.flutter.io/quick_actions'),
+  );
+
+  final MethodChannel _kChannel;
```
Comment (cyanglaz) [inline packages/quick_actions/lib/quick_actions.dart]: Let's keep the _KChannel as the constant for init by default and give this instance member a different name. We usually use a name started with k only for contants.
How about:
```suggestion
  final MethodChannel channel;
```

File: packages/quick_actions/test/quick_actions_test.dart [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  test('setShortcutItems with demo data', () async {
+    const String type = 'type';
+    const String localizedTitle = 'localizedTitle';
+    const String icon = 'icon';
+    await quickActions.setShortcutItems(
+      const <ShortcutItem>[
+        ShortcutItem(type: type, localizedTitle: localizedTitle, icon: icon)
+      ],
+    );
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
```
Comment (cyanglaz) [inline packages/quick_actions/test/quick_actions_test.dart]: Remove this TODO and add the type parameter. Same for below.

File: packages/quick_actions/lib/quick_actions.dart [hunk 4]
```diff
@@ -36,7 +33,16 @@ class ShortcutItem {
 
 /// Quick actions plugin.
 class QuickActions {
-  const QuickActions();
+  factory QuickActions() => _instance;
+
+  @visibleForTesting
+  QuickActions.private(MethodChannel channel) : _kChannel = channel;
```
Comment (cyanglaz) [inline packages/quick_actions/lib/quick_actions.dart]: It is not private. maybe:
```dart
QuickActions.withMethodChannel(MethodChannel channel) : _kChannel = channel;
```

**Comment by The-Redhat** [comment]
I made the requested changes

**Comment by collinjackson** [comment]
We discussed as a team and we'd like unit tests to be written using `setMockMethodCallHandler` rather than mockito.

Here's an example unit test:
https://github.com/flutter/plugins/blob/master/packages/cloud_functions/test/cloud_functions_test.dart

Could you please update your unit test to use this style instead?

**Comment by The-Redhat** [comment]
Ok. What do I use when I want to test that `setMethodCallHandler` is called ?

**Comment by The-Redhat** [comment]
@collinjackson I updated the tests

**Comment by The-Redhat** [comment]
@collinjackson Can I do something to get this pr merged ?

**Pull request merged**